.github/workflows/pytest.yml

```
@@ -0,0 +1,28 @@
+name: Run Pytest on Pull Request
+
+on:
+  pull_request:
+    branches:
+      - main  # Change this to your default branch if it's not 'main'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Run Pytest
+        run: |
+          python -m pytest -v tests/
```

```suggestion
          pytest -vs --cov modules --cov-report term-missing tests/
```
may need `pytest-cov` as additional dependency

---

running pytest yields "modules not found". hence the "python -m" approach. Still I can add the remaining parameters.